File: rides-service/src/main/java/io/simakkoi9/ridesservice/model/dto/kafka/KafkaDriverDto.java
package io.simakkoi9.ridesservice.model.dto.kafka;

import io.simakkoi9.ridesservice.model.entity.Gender;
import java.io.Serializable;

public record KafkaDriverDto(
    Long id,

    String name,

    String email,

    String phone,

    Gender gender,

    KafkaCarDto car
) implements Serializable {}

File: driver-service/src/main/kotlin/io/simakkoi9/driverservice/model/dto/kafka/KafkaDriverDto.kt
package io.simakkoi9.driverservice.model.dto.kafka

import io.simakkoi9.driverservice.model.entity.Gender

data class KafkaDriverDto(
    val id: Long,

    val name: String,

    val email: String,

    val phone: String,

    val gender: Gender,

    val car: KafkaCarDto
)